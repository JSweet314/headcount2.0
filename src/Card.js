import React from 'react';
import PropTypes from 'prop-types';
import './styles/Card.css';


const Card = ({ district }) => {
  return (
    <article className='district'>
      <h3>{ district.location } </h3>
      <ul>
        {
          Object.keys(district.stats).map( (timeFrame, index) => {
            return (
              <li 
                key={index}
                className={district.stats[timeFrame] < 0.5 ? 'red' : ''}
              >
                {timeFrame}:
                {Math.round(district.stats[timeFrame] * 100) / 100}
              </li> 
            );
          })    
        }
      </ul>
    </article>
  
  );
};

Card.propTypes = { 
  district: PropTypes.object.isRequired
};

export default Card;