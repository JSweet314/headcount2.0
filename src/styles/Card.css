.red {
  color: red;
}

li {
  color: green;
}