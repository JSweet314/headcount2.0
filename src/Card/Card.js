import React from 'react';
import PropTypes from 'prop-types';
import './Card.css';


const Card = ({ location, stats, selectLocation, selectedLocations }) => {
  const selected = selectedLocations.some(district => district.location === location) ? 'selected' : '';

  return (
    <article 
      className={`district ${selected}`}
      onClick={() => selectLocation(location)} >
      <h3>{location} </h3>
      <ul>
        {
          Object.keys(stats).map( (timeFrame, index) => {
            return (
              <li 
                key={index}
                className={stats[timeFrame] < 0.5 ? 'red' : ''}
              >
                {timeFrame}:
                {Math.round(stats[timeFrame] * 100) / 100}
              </li> 
            );
          })    
        }
      </ul>
    </article>
  
  );
};

Card.propTypes = { 
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired,
  selectLocation: PropTypes.func.isRequired,
  selectedLocations: PropTypes.array.isRequired
};

export default Card;