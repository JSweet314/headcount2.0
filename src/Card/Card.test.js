import React from 'react';
import ReactDOM from 'react-DOM';
import Card from './Card';
import {shallow} from 'enzyme';

describe('Card', () => {
	let card, district, stats;
	const selectLocation = jest.fn();
	const selectedLocations = ['COLORADO', 'COLORADO SPRINGS'];

	beforeEach(() => { 
		stats = { '2017': 0 }
		card = shallow(<Card 
			location="COLORADO"
			stats={stats} 
			selectLocation={selectLocation} 
			selectedLocations={selectedLocations} />);
	});

	it('renders the article' , () => {
		expect(card).toMatchSnapshot();
	});

	it('has props district which is an object', () => {
		expect(card.props("district")).toBeDefined();
	});

	it('has a class "red" if district percentile is less than 0.5', () => {
		expect(card.find('.red').length).toEqual(1);
		stats['2017'] = 0.51;
		card = shallow(<Card
			location="COLORADO"
			stats={stats} 
			selectLocation={selectLocation} 
			selectedLocations={selectedLocations} />);
		expect(card.find('.red').length).toEqual(0);
	});

})