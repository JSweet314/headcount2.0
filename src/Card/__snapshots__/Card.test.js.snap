// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    location="COLORADO"
    selectLocation={[Function]}
    selectedLocations={
        Array [
            Object {
              "location": "COLORADO",
              "stats": Object {
                "2016": 0.45,
                "2017": 0.34,
              },
            },
            Object {
              "location": "COLORADO SPRINGS",
              "stats": Object {
                "2016": 0.45,
                "2017": 0.34,
              },
            },
          ]
    }
    stats={
        Object {
            "2017": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h3>
                    COLORADO
          </h3>
</header>,
        <ul>
          <li
                    className="red"
          >
                    2017
                    : 
                    0
                     %
          </li>
</ul>,
      ],
      "className": "district selected",
      "id": undefined,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h3>
            COLORADO
</h3>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COLORADO",
          },
          "ref": null,
          "rendered": "COLORADO",
          "type": "h3",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="red"
>
              2017
              : 
              0
               %
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "2017",
                ": ",
                0,
                " %",
              ],
              "className": "red",
            },
            "ref": null,
            "rendered": Array [
              "2017",
              ": ",
              0,
              " %",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h3>
                        COLORADO
            </h3>
</header>,
          <ul>
            <li
                        className="red"
            >
                        2017
                        : 
                        0
                         %
            </li>
</ul>,
        ],
        "className": "district selected",
        "id": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h3>
              COLORADO
</h3>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "COLORADO",
            },
            "ref": null,
            "rendered": "COLORADO",
            "type": "h3",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="red"
>
                2017
                : 
                0
                 %
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "2017",
                  ": ",
                  0,
                  " %",
                ],
                "className": "red",
              },
              "ref": null,
              "rendered": Array [
                "2017",
                ": ",
                0,
                " %",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
